src/solver/exprtype.go

```diff
@@ -202,6 +202,41 @@ func binaryPlusOpType(sc *meta.Scope, cs *meta.ClassParseState, left, right ir.N
 	return binaryMathOpType(sc, cs, left, right, custom)
 }
 
+func classNameToString(cs *meta.ClassParseState, n ir.Node) (string, bool) {
+	var name string
+
+	switch n := n.(type) {
+	case *ir.String:
+		name = n.Value
+	case *ir.ClassConstFetchExpr:
+		if !strings.EqualFold(n.ConstantName.Value, "class") {
+			return "", false
+		}
+
+		switch class := n.Class.(type) {
+		case *ir.Name:
+			name = class.Value
+		case *ir.Identifier:
+			name = class.Value
+		case *ir.SimpleVar:
+			name = "$" + class.Name
+		default:
+			return "", false
+		}
+	default:
+		return "", false
+	}
+
+	className, ok := GetClassName(cs, &ir.Name{
+		Value: name,
+	})
```

Comment:
Can this literal be placed on the same line as GetClassName call?
```go
	className, ok := GetClassName(cs, &ir.Name{Value: name})
```

Reply:
Done